### PR TITLE
fixes : Test Coverage for Azure KeyVault Keys

### PR DESCRIPTION
**Pull request by ksatwara**
#6485

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java [hunk 1]
```diff
@@ -80,6 +80,16 @@ public void createRsaKey(HttpClient httpClient, KeyServiceVersion serviceVersion
...
     }
 
+    /**
+     * Tests that a Ec key created.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void createEcKey(HttpClient httpClient, KeyServiceVersion keyServiceVersion) {
+        createKeyAsyncClient(httpClient, keyServiceVersion);
+        createEcKeyRunner((expected) -> client.createEcKey(expected));
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java:90]: It is important that we assert that the result of the operation is correct, so I would at least add one assertion like on the method above (`createRsaKey`).

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java [hunk 2]
```diff
@@ -76,6 +76,16 @@ public void createRsaKey(HttpClient httpClient, KeyServiceVersion serviceVersion
...
     }
 
+    /**
+     * Tests that an Ec key is created.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void createEcKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
+        getKeyClient(httpClient, serviceVersion);
+        createEcKeyRunner((expected) -> client.createEcKey(expected));
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java:86]: Same comment as above about assertions.

File: sdk/keyvault/azure-security-keyvault-keys/src/test/resources/session-records/createEcKey.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      "Cache-Control" : "no-cache",
+      "X-AspNet-Version" : "4.0.30319",
+      "x-ms-keyvault-region" : "westus",
+      "x-ms-keyvault-network-info" : "conn_type=Ipv4;addr=174.127.169.154;act_addr_fam=InterNetwork;",
+      "Expires" : "-1",
+      "Content-Length" : "715",
+      "x-ms-request-id" : "37285ec5-78a9-4eb3-b1c9-a042721b47e5",
+      "x-ms-keyvault-service-version" : "1.1.10.0",
+      "Body" : "{\"key\":{\"kid\":\"https://azure-kv-tests2.vault.azure.net/keys/javaKeyTemp/dfa0588bcdb74b84954421a69260882d\",\"kty\":\"RSA\",\"key_ops\":[\"encrypt\",\"decrypt\",\"sign\",\"verify\",\"wrapKey\",\"unwrapKey\"],\"n\":\"12GE3dE0RBZoo2lF3yCBpr6yKUt1zq9ImepwD6Ee_HeQE9P2AcnZD4hK4x-zHtCRSKoq0SxsKKF5svfk2FD7oQYmAv53zXhenzJyfk698Za9liUJMKb11cPtGhIPuANlcDbwroPrrT-d2A9tW2-sttorBhg8Rud_s9vl6VaeeCJ3Z5SwApoLVn3MJCzMmtYuRwfEL6wKOizmmUqTUTk5kRlrR8AdZ5U9-BqkIfRW_tkr8_3rgLA6qKcT2uFUiXD_UJskULNZQzUXFnMi4OFSgIcocPdXm1cBe5FL6llMYFJIjYCAfaHF89v_tokyW9rKQzhXV044jaKKwq9JGsNTow\",\"e\":\"AQAB\"},\"attributes\":{\"enabled\":true,\"nbf\":949237199,\"exp\":2527113600,\"created\":1596509358,\"updated\":1596509358,\"recoveryLevel\":\"Recoverable+Purgeable\",\"recoverableDays\":90}}",
```
[the remark this answers is not in the document]
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-keys/src/test/resources/session-records/createEcKey.json:24]: You will probably need to run a live test with a real Key Vault and record the operation. I can help you out if you have questions about how to do this :)

[the remark this answers is not in the document]
Comment (ksatwara) [reply sdk/keyvault/azure-security-keyvault-keys/src/test/resources/session-records/createEcKey.json:24]: @vcolin7 Thanks for the guide but i don't have live testing project so could you please assign me test project so i will work on that or is there any testing project which you can assign me so work more on that

[the remark this answers is not in the document]
Comment (ksatwara) [reply sdk/keyvault/azure-security-keyvault-keys/src/test/resources/session-records/createEcKey.json]: moreover I also message to @joshfree about the same on Gitter but still no reply so i am asking you for the same.

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java [hunk 4]
```diff
@@ -80,6 +80,16 @@ public void createRsaKey(HttpClient httpClient, KeyServiceVersion serviceVersion
...
     }
 
+    /**
+     * Tests that a Ec key created.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void createEcKey(HttpClient httpClient, KeyServiceVersion keyServiceVersion) {
+        createKeyAsyncClient(httpClient, keyServiceVersion);
+        createEcKeyRunner((expected) -> assertNotNull(client.createEcKey(expected)));
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java:90]: I think we can use `assertKeyEquals` like the method above to verify the created key contains the same characteristics we sent in the payload.

```suggestion
        createEcKeyRunner((expected) -> StepVerifier.create(client.createEcKey(expected))
            .assertNext(response -> assertKeyEquals(expected, response))
            .verifyComplete());
```

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java [hunk 5]
```diff
@@ -76,6 +76,16 @@ public void createRsaKey(HttpClient httpClient, KeyServiceVersion serviceVersion
...
     }
 
+    /**
+     * Tests that an Ec key is created.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void createEcKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
+        getKeyClient(httpClient, serviceVersion);
+        createEcKeyRunner((expected) -> assertNotNull(client.createEcKey(expected)));
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java:86]: Same comment as above :)

```suggestion
        createEcKeyRunner((expected) -> StepVerifier.create(client.createEcKey(expected))
            .assertNext(response -> assertKeyEquals(expected, response))
            .verifyComplete());
```

**Review by vcolin7 — CHANGES_REQUESTED** [review]

File: sdk/keyvault/azure-security-keyvault-keys/src/test/resources/session-records/createEcKey.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-keys/src/test/resources/session-records/createEcKey.json]: As far as I know we do not provide live applications for the community to test on, however, it you can [create an Azure Subscription for free](https://azure.microsoft.com/en-us/services/key-vault/), as well as an [Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/keys/quick-create-portal). You can find more details on how to get started [here](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/keyvault/azure-security-keyvault-keys#getting-started) and [here](https://docs.microsoft.com/en-us/azure/key-vault/secrets/quick-create-java).

If you wish to record tests yourself, you would need to pass the details of your Azure Key Vault as environment variables with the following names: `ARM_CLIENTID`, `ARM_CLIENTKEY` and `AZURE_TENANT_ID`. You will also need to set the `AZURE_TEST_MODE` environment variable to `RECORD`. After running the desired test, the recording can be found in the `target/session-records` as a file with the following name: `<test-name>.json`. I can also give you a hand if you get stuck at some point of the process.

As an alternative, I can record the test myself and push the recording to your branch :)

File: sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java [hunk 6]
```diff
@@ -76,6 +76,16 @@ public void createRsaKey(HttpClient httpClient, KeyServiceVersion serviceVersion
...
     }
 
+    /**
+     * Tests that an Ec key is created.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void createEcKey(HttpClient httpClient, KeyServiceVersion serviceVersion) {
+        getKeyClient(httpClient, serviceVersion);
+        createEcKeyRunner((expected) -> assertKeyEquals(expected, client.createEcKey(expected)));
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyClientTest.java:86]: We should do the same for both clients.

```suggestion
        createEcKeyRunner((expected) -> StepVerifier.create(client.createEcKey(expected))
            .assertNext(response -> assertKeyEquals(expected, response))
            .verifyComplete());
```

**Review by vcolin7 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**